python/dgl/sparse/elementwise_op_sp.py

```diff
@@ -150,6 +163,12 @@ def sp_div(A: SparseMatrix, B: Scalar) -> SparseMatrix:
     """
     if is_scalar(B):
         return val_like(A, A.val / B)
+    if A.is_diag() and B.is_diag():
+        assert A.shape == B.shape, (
+            f"The shape of diagonal matrix A {A.shape} and B {B.shape} must"
+            f"match for elementwise division."
```

Comment:
@frozenbugs I wonder if we will have some guidance on how to write error messages. Shape checks are particularly common.

Reply:
I am fine if we just use the same pattern, in the future code review, we should just ask ppl to follow this patter.

It is probably not worth to create a shared function / string for this.